src/sqlfluff/core/parser/grammar/base.py

```diff
@@ -503,35 +538,52 @@ def _bracket_sensitive_look_ahead_match(cls, segments, matchers, parse_context):
                     # that we're otherwise looking for.
                     pre, match, matcher = cls._look_ahead_match(
                         seg_buff,
-                        matchers,
+                        matchers + bracket_matchers,
                         parse_context=parse_context,
                     )
 
                     if match:
-                        if matcher in start_brackets:
+                        if matcher in matchers:
+                            # It's one of the things we were looking for!
+                            # Return.
+                            return (pre_seg_buff + pre, match, matcher)
+                        elif matcher in start_brackets:
                             # We've found the start of a bracket segment.
                             # NB: It might not *Actually* be the bracket itself,
                             # but could be some non-code element preceding it.
                             # That's actually ok.
 
                             # Add the bracket to the stack.
                             bracket_stack.append(match.matched_segments[0])
+                            definite_stack.append(
+                                start_definite[start_brackets.index(matcher)]
+                            )
                             # Add the matched elements and anything before it to the
                             # pre segment buffer. Reset the working buffer.
                             pre_seg_buff += pre
                             pre_seg_buff += match.matched_segments
                             seg_buff = match.unmatched_segments
                             continue
                         elif matcher in end_brackets:
-                            # We've found an unexpected end bracket!
-                            raise SQLParseError(
-                                "Found unexpected end bracket!",
-                                segment=match.matched_segments[0],
-                            )
+                            # each bracket with its "definite" attribute
+                            bracket_is_definite = end_definite[
+                                end_brackets.index(matcher)
+                            ]
+                            if bracket_is_definite:

```

Comment:
I'd like @alanmcruickshank 's stamp of approval here because this bracket logic is new to me, and I added this condition to ignore unexpected end brackets in case the bracket is not definite.

This allows us in BigQuery to reuse `<`, and `>` as brackets without parsing them as different keywords (they're still LowerThan and GreaterThan). And not throw exceptions when they're used as comparison operators.

Reply:
I agree about getting Alan's review. I think he wrote all the bracket changes for this PR, other than your final changes, Daniel, but at the time, it was more in the spirit of "can we get this working" rather than "is this production quality". I think I mentioned somewhere in this PR, I wonder if the code as written will report bracket-match-related parse errors in a clean and standard way.